### ISSUE TITLE
Why the database saved by KeePassDX have different file size compared to the PC KeePass saved? 

### ISSUE THREAD

**Issue by HaujetZhao** [Kunzisoft/KeePassDX#1016]
**Describe the bug**

I am using Syncthing to sync the database between my phone and my PC, and I noticed that, every time I saved the database by KeepassDX, it's size become about 1.7MB, and every time I saved the database by PC KeePass, it's size become about 3.51MB, why would this happen? 

**Expected behavior**
Using same alogrism, I suppose they should have size after saving. 

**KeePass Database**
 - Created with: Windows KeePass 2.44
 - Version: 2
 - Location: local file
 - Size: 3.51MB
 - Contains attachment: Yes, 4 pictures, but all destructed, don't know why, don't know how, don't know when it happened. I'll upload one of the destructed attatchments. 
 - [Destructed Picture Attatchment](https://user-images.githubusercontent.com/19181833/121110135-5df89880-c83f-11eb-8ed0-021c732f073a.jpg)

**KeePassDX (please complete the following information):**
 - Version: 2.9.18
 - Build: Free
 - Language: Chinese Simplified

**Android (please complete the following information):**
 - Device: Redmi K20 Pro
 - Version: 11

**Comment by J-Jamet** [comment]
There can be two explanations
1. The third-party applications you use to synchronize files do not update caches and backups are not visible. In this case, you should contact the developers of these other applications to find solutions.
2. Maybe a bug in KeePassDX that prevents from formatting correctly the binary format of the attached files. 

Can you upload a test database to reproduce the problem? Have you tried to recover the file manually to see if KeePassDX was saving correctly with a local file manager? Because I did the tests on my side and I have no problem.

**Comment by J-Jamet** [comment]
Of course, your KeePassDX application must be up to date: **2.10.2**

**Comment by HaujetZhao** [comment]
KeepassDX updated to the 2.10.2

Each time sync, Syncthing will make a copy of old file as a backup. 

Here is the test vault that can reproduce this issue: 

[Test Vault.zip](https://github.com/Kunzisoft/KeePassDX/files/6620332/Test.Vault.zip)

**Comment by J-Jamet** [comment]
I can't test your test bases without the associated passwords.

**Comment by J-Jamet** [comment]
> Each time sync, Syncthing will make a copy of old file as a backup.

If this is not the expected behavior, open a ticket on the Syncthink project.

**Comment by HaujetZhao** [comment]
> I can't test your test bases without the associated passwords.

Oh, sorry!!! Forgot to mention, the password is `happy`

**Comment by HaujetZhao** [comment]
@J-Jamet  Just wanna check it out, did this reproduced with the database I uploaded?

**Comment by J-Jamet** [comment]
No, can't reproduce. You don't indicate the entry that contains the attachments, there are many of them. Note that there is a history to your base. And for you ? the problem is reproducible since you updated the application?

**Comment by HaujetZhao** [comment]
Ah ah, forgot the history...A Lot of work then. 

I'll record a video later.

**Comment by HaujetZhao** [comment]
Have this new created vault as an example, it's saved by Windows Keepass 2.44, it's size is 509KB (contains 2 attachments)  [test vault.zip](https://github.com/Kunzisoft/KeePassDX/files/6645448/test.vault.zip), password is `happy`

But if you save it with KeepassDX, it's size will become 287KB.

**Comment by J-Jamet** [comment]
I just checked and I don't see any problem.
The attached files are not altered during the backup (binaries hashes remain the same) and the file formatting is repaired and ordered to take less space.
https://github.com/Kunzisoft/KeePassDX/wiki/FAQ#why-is-the-database-file-smaller-after-a-save

**Comment by h-h-** [comment]
This still occurs.
DB stored with KeePassXC 2.7.9 on Win10: **152KB**,
edited with KeePassDX from F-Droid on Android 4.0.8: **129KB**
Exported both to xml and only find expected differences.
